module4/homework/appendNewRecipe/AppendNewRecipe.cpp

```diff
@@ -0,0 +1,60 @@
+#include "AppendNewRecipe.hpp"
+#include <algorithm>
+#include <fstream>
+#include <iterator>
+#include <sstream>
+
+std::vector<std::string> FormatIngredients(
+    const std::list<std::string>& ingredients,
+    const std::deque<std::pair<size_t, char>>& amount) {
+    std::vector<std::string> formattedIngredents{};
+
+    std::transform(
+        ingredients.cbegin(), ingredients.cend(), amount.cbegin(),
+        std::back_inserter(formattedIngredents),
+        [](const std::string ing, const std::pair<size_t, char> amount) -> std::string {
+            switch (amount.second) {
+            case 'g':
```

Comment:
I think in this case is better to use map where: 'g'/etc = key, "gram" = value. However, your solution is good; p

Reply:
Agree map will be shorter.